internal/app/machined/pkg/controllers/cluster/local_affiliate.go

```diff
@@ -108,188 +114,204 @@ func (ctrl *LocalAffiliateController) Outputs() []controller.Output {
-				}
+			continue
+		}
 
-				continue
+		hostname, err := safe.ReaderGetByID[*network.HostnameStatus](ctx, r, network.HostnameID)
+		if err != nil {
+			if !state.IsNotFoundError(err) {
+				return fmt.Errorf("error getting hostname: %w", err)
 			}
 
-			nodename, err := safe.ReaderGetByID[*k8s.Nodename](ctx, r, k8s.NodenameID)
-			if err != nil {
-				if !state.IsNotFoundError(err) {
-					return fmt.Errorf("error getting nodename: %w", err)
-				}
+			continue
+		}
 
-				continue
+		nodename, err := safe.ReaderGetByID[*k8s.Nodename](ctx, r, k8s.NodenameID)
+		if err != nil {
+			if !state.IsNotFoundError(err) {
+				return fmt.Errorf("error getting nodename: %w", err)
 			}
 
-			routedAddresses, err := safe.ReaderGetByID[*network.NodeAddress](ctx, r, network.FilteredNodeAddressID(network.NodeAddressRoutedID, k8s.NodeAddressFilterNoK8s))
-			if err != nil {
-				if !state.IsNotFoundError(err) {
-					return fmt.Errorf("error getting addresses: %w", err)
-				}
+			continue
+		}
 
-				continue
+		routedAddresses, err := safe.ReaderGetByID[*network.NodeAddress](ctx, r, network.FilteredNodeAddressID(network.NodeAddressRoutedID, k8s.NodeAddressFilterNoK8s))
+		if err != nil {
+			if !state.IsNotFoundError(err) {
+				return fmt.Errorf("error getting addresses: %w", err)
 			}
 
-			currentAddresses, err := safe.ReaderGetByID[*network.NodeAddress](ctx, r, network.FilteredNodeAddressID(network.NodeAddressCurrentID, k8s.NodeAddressFilterNoK8s))
-			if err != nil {
-				if !state.IsNotFoundError(err) {
-					return fmt.Errorf("error getting addresses: %w", err)
-				}
+			continue
+		}
 
-				continue
+		currentAddresses, err := safe.ReaderGetByID[*network.NodeAddress](ctx, r, network.FilteredNodeAddressID(network.NodeAddressCurrentID, k8s.NodeAddressFilterNoK8s))
+		if err != nil {
+			if !state.IsNotFoundError(err) {
+				return fmt.Errorf("error getting addresses: %w", err)
 			}
 
-			machineType, err := safe.ReaderGetByID[*config.MachineType](ctx, r, config.MachineTypeID)
-			if err != nil {
-				if !state.IsNotFoundError(err) {
-					return fmt.Errorf("error getting machine type: %w", err)
-				}
+			continue
+		}
 
-				continue
+		machineType, err := safe.ReaderGetByID[*config.MachineType](ctx, r, config.MachineTypeID)
+		if err != nil {
+			if !state.IsNotFoundError(err) {
+				return fmt.Errorf("error getting machine type: %w", err)
 			}
 
-			// optional resources (kubespan)
-			kubespanIdentity, err := safe.ReaderGetByID[*kubespan.Identity](ctx, r, kubespan.LocalIdentity)
-			if err != nil && !state.IsNotFoundError(err) {
-				return fmt.Errorf("error getting kubespan identity: %w", err)
-			}
+			continue
+		}
 
-			kubespanConfig, err := safe.ReaderGetByID[*kubespan.Config](ctx, r, kubespan.ConfigID)
-			if err != nil && !state.IsNotFoundError(err) {
-				return fmt.Errorf("error getting kubespan config: %w", err)
+		machineConfig, err := safe.ReaderGetByID[*config.MachineConfig](ctx, r, config.V1Alpha1ID)
```

Comment:
I'm not sure we should _require_ machine config here - we can still generate a local affiliate, but we can't fill only local API server port if we don't have it.

Reply:
i think this is still not resolved, correct?